src/components/SpendingLimit/index.tsx

```diff
@@ -0,0 +1,92 @@
+/** @jsxImportSource @emotion/react */
+import BigNumber from 'bignumber.js';
+import { VError, formatVErrorToReadableString } from 'errors';
+import React from 'react';
+import { useTranslation } from 'translation';
+import { Token } from 'types';
+
+import { LabeledInlineContent } from 'components/LabeledInlineContent';
+import { Spinner } from 'components/Spinner';
+import { Tooltip } from 'components/Tooltip';
+import useFormatTokensToReadableValue from 'hooks/useFormatTokensToReadableValue';
+
+import { TextButton } from '../Button';
+import { Icon } from '../Icon';
+import { toast } from '../Toast';
+import { useStyles } from './styles';
+
+export interface SpendingLimitProps {
+  token: Token;
+  onRevoke: () => Promise<unknown>;
+  isRevokeLoading: boolean;
+  walletBalanceTokens?: BigNumber;
+  walletSpendingLimitTokens?: BigNumber;
+  className?: string;
+}
+
+export const SpendingLimit: React.FC<SpendingLimitProps> = ({
+  walletSpendingLimitTokens,
+  walletBalanceTokens,
+  isRevokeLoading,
+  token,
+  onRevoke,
+  ...otherContainerProps
+}) => {
+  const { t } = useTranslation();
+  const styles = useStyles();
+
+  const handleRevoke = async () => {
+    try {
+      await onRevoke();
+    } catch (error) {
+      let { message } = error as Error;
+
+      if (error instanceof VError) {
+        message = formatVErrorToReadableString(error);
+      }
```

Comment:
Since message isn't guaranteed to be user friendly you could refactor this to always show the user the value from `formatVErrorToReadableString` and return a generic message if the error can't be parsed.

```suggestion
        const message = formatErrorToReadableString(error);
```

Reply:
It makes sense from a user standpoint, however at the moment our error handling isn't good enough to show generic error message: we just disabled Sentry so at the moment we depend on users to report errors to us, which they do through screenshots... It's dirty, but hopefully we can activate Sentry again soon and improve our error handling.